Do not ship contrib .gitignores

In [bundled versions](http://premake.github.io/download.html#v5) of Premake 5, .gitignores are included. The top-level one is not harmful (and might be needed if I understood correctly issue #595) but there are some of those files under contrib/ that can cause problems.

Namely, `contrib/curl/include/curl/.gitignore` excludes `contrib/curl/include/curl/curlbuild.h` which is needed to build premake5. When including the bundled version of premake in a git repository, this file is silently ignored and users stumble upon this kind of failure:
```
==== Building curl-lib (release) ====
asyn-thread.c
In file included from ../../contrib/curl/lib/asyn-thread.c:60:
In file included from ../../contrib/curl/lib/urldata.h:82:
In file included from ../../contrib/curl/lib/cookie.h:26:
../../contrib/curl/include/curl/curl.h:34:10: fatal error: 'curlbuild.h' file not found
#include "curlbuild.h"       /* libcurl build definitions */
         ^
1 error generated.
```

I think those third-party gitignore files should be removed from bundles.
```
contrib/curl/include/curl/.gitignore
contrib/curl/lib/.gitignore
contrib/mbedtls/.gitignore
contrib/mbedtls/{include,library,programs,test,yotta}/.gitignore
```

--- a/scripts/package.lua
+++ b/scripts/package.lua
@@ -127,17 +127,15 @@
 
 	print("Cleaning up the source tree...")
 	os.rmdir("packages")
+	os.rmdir(".git")
 
-	local modules = table.join(".", os.matchdirs("modules/*"))
-	for _, module in ipairs(modules) do
-		for _, name in ipairs { ".git" } do
-			os.rmdir(path.join(module, name))
-		end
-		for _, name in ipairs { ".DS_Store", ".git", ".gitignore", ".gitmodules", ".travis.yml", ".editorconfig", "Bootstrap.mak" } do
-			os.remove(path.join(module, name))
+	local removelist = { ".DS_Store", ".git", ".gitignore", ".gitmodules", ".travis.yml", ".editorconfig", "appveyor.yml", "Bootstrap.mak" }
+	for _, removeitem in ipairs(removelist) do
+		local founditems = os.matchfiles("**" .. removeitem)
+		for _, item in ipairs(founditems) do
+			os.remove(item)
 		end
 	end
-
 
 --
 -- Generate a source package.
@@ -152,9 +150,11 @@ if kind == "source" then
 	execQuiet("premake5 /to=build/vs2012 vs2012")
 	execQuiet("premake5 /to=build/vs2013 vs2013")
 	execQuiet("premake5 /to=build/vs2015 vs2015")
+	execQuiet("premake5 /to=build/vs2017 vs2017")
 	execQuiet("premake5 /to=build/gmake.windows /os=windows gmake")
 	execQuiet("premake5 /to=build/gmake.unix /os=linux gmake")
 	execQuiet("premake5 /to=build/gmake.macosx /os=macosx gmake")
+	execQuiet("premake5 /to=build/gmake.bsd /os=bsd gmake")
 
 	print("Creating source code package...")
 	os.chdir("..")

